CMakeLists.txt

```diff
@@ -151,9 +151,7 @@ if($ENV{ENABLE_TRACY})
     include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/tracy.cmake)
 endif()
 
-# Build umd_device
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/umd_device.cmake)
-
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/tt_metal/third_party/umd)
```

Comment:
can we add UMD via `ExternalProject` ? This avoids any potential namespace pollution

Reply:
Can you help me understand the problem you want to avoid with this?
I often edit code in `tt_metal/third_party/umd` and want the build to react appropriately in this situation.

Reply:
Could we make this behaviour opt in with a `-D` option at cmake first execution time?

Reply:
https://inhzus.io/posts/2023-12-01-cmake-external-project/#intro

you can test with `add_subdirectory`, but switch to `ExternalProject_Add` when you check in